packages/flutter_tools/lib/src/reporting/unified_analytics.dart

```diff
@@ -50,3 +52,13 @@ Analytics getAnalytics({
     enableAsserts: enableAsserts,
   );
 }
+
+/// Function to safely grab the max rss from [ProcessInfo].
+int? getMaxRss(ProcessInfo processInfo) {
+  try {
+    return globals.processInfo.maxRss;
+  } on Exception catch (error) {
+    globals.printTrace('Querying maxRss failed with error: $error');
```

Comment:
if this fails consistently, we'd print this twice, right (since we still calculate the legacy event)? Can you instead do this calculation once, and pass it to both analytics events?